### PR TITLE
Md5challenge

### PR DESCRIPTION
**Pull request by commjoen**
Thank you for submitting a pull request to the WrongSecrets app!

### CODE REVIEW

**Comment by commjoen** [comment]
Hosted at https://arcane-scrubland-42646.herokuapp.com/challenge/18 for testing.

File: src/main/resources/explanations/challenge18_reason.adoc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+*Why MD5 and SHA1 hashing alone are not enough.*
+
+MD5 and SHA1 hash are no longer considered safe to store passwords on their own. Speed is what makes MD5 and SHA1 hashes so useful, but it is also their downfall. It only takes a few minutes to hash thousands of passwords; this also means that it only takes minutes to hash thousands of common passwords and use these hashes to compare against a hash that has been obtained.
+
+Companies try different techniques to harden MD5 and SHA1 hashes, such as "salting" them. This is the process of adding additional characters to the password that only the person/company that should be decrypting knows. Unfortunately this is not enough either with the raise of GPU and ASIC based computations. Therefore, companies using these techniques can better migrate to https://en.wikipedia.org/wiki/Argon2[Argon2] or https://en.wikipedia.org/wiki/Balloon_hashing[Balloon hashing].
```
Comment (bendehaan) [inline src/main/resources/explanations/challenge18_reason.adoc:5]:
```suggestion
Companies try different techniques to harden MD5 and SHA1 hashes, such as "salting" them. This is the process of adding additional characters to the password that only the person/company that should be decrypting knows. Unfortunately this is not enough either with the rise of GPU and ASIC based computations. Therefore, companies using these techniques can better migrate to https://en.wikipedia.org/wiki/Argon2[Argon2] or https://en.wikipedia.org/wiki/Balloon_hashing[Balloon hashing].
```

File: src/main/resources/explanations/challenge18_hint.adoc [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+You can solve this challenge using the following steps:
+
+1. For the first hash (MD5):
+- Use a tool such as Hashcat :
+- Install https://hashcat.net/hashcat/[Hashcat]
+- Download the https://github.com/brannondorsey/naive-hashcat/releases/download/data/rockyou.txt[rockyou.txt password list]
+- Run Hashcat on the hash `hashcat -m 0 "2ab96390c7dbe3439de74d0c9b0b1767" /path/to/file/rockyou.txt` 
+
+2. For the second hash (SHA1):
+- Use a tool such as Hashcat :
```
Comment (bendehaan) [inline src/main/resources/explanations/challenge18_hint.adoc:12]:
```suggestion
- Use a tool such as Hashcat:
```

File: src/main/resources/explanations/challenge18_hint.adoc [hunk 3]
```diff
@@ -0,0 +1,20 @@
+This challenge is specifically looking at MD5 and SHA1 hashes without salting. Are these un-crackable?
+
+You can solve this challenge using the following steps:
+
+1. For the first hash (MD5):
+- Use a tool such as Hashcat :
```
Comment (bendehaan) [inline src/main/resources/explanations/challenge18_hint.adoc:6]:
```suggestion
- Use a tool such as Hashcat:
```

File: src/main/resources/explanations/challenge18_reason.adoc [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+As a user you often have no choice in how your passwords are stored; the only thing you can do in this case is try to make your password longer and more complex. A password SHA1 that is 7 characters long with upper and lowercase characters will take roughly a minute or 5 to brute force on a proper GPU, whereas one with 25 characters will take much longer. That would take some real patience.
```
Comment (bendehaan) [inline src/main/resources/explanations/challenge18_reason.adoc:7]:
```suggestion
As a user you often have no choice in how your passwords are stored; the only thing you can do in this case is try to make your password longer and more complex. A password SHA1 that is 7 characters long with upper and lowercase characters will take roughly a minute or 5 to brute force on a proper GPU, whereas one with 25 characters will take much longer.
```

**Comment by bendehaan** [comment]
Works nicely, just some textual nits

**Review by bendehaan — APPROVED** [review]

**Pull request merged**